### ISSUE TITLE
validate birthday

### ISSUE THREAD

**Issue by marcel-liebgott** [script-solution/Boardsolution#83]
Validate the birthday for their valid.
I have a registrated user and his date of birth is in the future. His age is -5. ;)

**Comment by hrniels** [comment]
Well, that would be nice, but the problem is that this is an additional profile field. Thus, currently we only have the option to specify a regular expression for validation, which does not suffice in this case. I'm not sure if it is important enough to either make the birthday a hardcoded field or extent the additional fields to support e.g. calling a user-defined function for validation.

**Comment by Rafioso** [comment]
@hrniels I do not agree with you, that this is an additional profile field, because it is in the standard installation of BS and you can not delete this profile field. You can only disable it. So even the user can't change the name of this profilefield (the hardcoded one) it should be checked like @marcel-liebgott said.

**Comment by hrniels** [comment]
Are we talking about the same thing here? Marcel suggested to validate the birthday in the sense that it's a valid date and in the past. I don't understand why it's relevant that it's part of the standard installation or whether you can delete or disable it.

I think the only question is as I stated above: do we want to make it a hardcoded field or extend the additional-field-feature to support user-defined functions or similar. Or we leave it as it is if it's not important enough.

**Comment by Rafioso** [comment]
> 
> 
> Are we talking about the same thing here? 

Yes i think and hope so. But i think, until we three are germans, the native language would be better to understand :-)

> 
> 
> I think the only question is as I stated above: do we want to make it a hardcoded field or extend the additional-field-feature to support user-defined functions or similar. Or we leave it as it is if it's not important enough.

IMHO it is not an additional-field, because you can not delete it in the ACP so why not hardcoded or why not really additional?

**Comment by hrniels** [comment]
> > Are we talking about the same thing here?
> 
> Yes i think and hope so. But i think, until we three are germans, the native language would be better to understand :-)

Ok, dann wechseln wir mal auf deutsch, ist vielleicht wirklich besser um Missverständnisse zu vermeiden :)

> > I think the only question is as I stated above: do we want to make it a hardcoded field or extend the additional-field-feature to support user-defined functions or similar. Or we leave it as it is if it's not important enough.
> 
> IMHO it is not an additional-field, because you can not delete it in the ACP so why not hardcoded or why not really additional?

Es ist zwar ein spezielles zusätzliches Profilfeld, weil man es nicht löschen kann, aber es wird mit der allgemeinen Infrastruktur realisiert. D.h. es gibt keinen speziellen Code für das Geburtstagsfeld. Daraus ergeben sich drei Möglichkeiten:

1. Wir erweitern die allgemeine Infrastruktur für die zusätzlichen Profilfelder, um solche Prüfungen beim Geburtstag und allgemein zu erlauben.
2. Wir nehmen den Geburtstag da raus und fügen überall speziellen Code dafür ein, der dann tun kann was er will.
3. Wir lassen alles wie es ist.

**Comment by Rafioso** [comment]
> 
> 
> > > Are we talking about the same thing here?
> > 
> > 
> > Yes i think and hope so. But i think, until we three are germans, the native language would be better to understand :-)
> 
> Ok, dann wechseln wir mal auf deutsch, ist vielleicht wirklich besser um Missverständnisse zu vermeiden :)


Der Sprachwechsel war jetzt eigentlich nur für dieses Thema gedacht *g*

> 
> > > I think the only question is as I stated above: do we want to make it a hardcoded field or extend the additional-field-feature to support user-defined functions or similar. Or we leave it as it is if it's not important enough.
> > 
> > 
> > IMHO it is not an additional-field, because you can not delete it in the ACP so why not hardcoded or why not really additional?
> 
> Es ist zwar ein spezielles zusätzliches Profilfeld, weil man es nicht löschen kann, aber es wird mit der allgemeinen Infrastruktur realisiert. D.h. es gibt keinen speziellen Code für das Geburtstagsfeld. Daraus ergeben sich drei Möglichkeiten:
> 
>     1. Wir erweitern die allgemeine Infrastruktur für die zusätzlichen Profilfelder, um solche Prüfungen beim Geburtstag und allgemein zu erlauben.
> 
>     2. Wir nehmen den Geburtstag da raus und fügen überall speziellen Code dafür ein, der dann tun kann was er will.
> 
>     3. Wir lassen alles wie es ist.

Punkt 2 oder 3 sollte es sein.

**Comment by marcel-liebgott** [comment]
Oh, da hab ich ja was angestellt 😅
